src/scores/probability/brier_impl.py

```diff
@@ -21,47 +25,205 @@ def brier_score(
+            # The .values is required as item() is not yet a valid method on Dask
+            if fcst.to_array().max().values.item() > 1 or fcst.to_array().min().values.item() < 0:
+                raise error_msg
+        else:
+            if fcst.max().values.item() > 1 or fcst.min().values.item() < 0:
+                raise error_msg
+        check_binary(obs, "obs")
+
+    return mse(fcst, obs, reduce_dims=reduce_dims, preserve_dims=preserve_dims, weights=weights)
+
+
+def brier_score_for_ensemble(
+    fcst: XarrayLike,  # type: ignore
+    obs: XarrayLike,  # type: ignore
+    ensemble_member_dim: str,
+    event_thresholds: Union[float, int, Sequence[float]],
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+    fair_correction: bool = True,
+    event_threshold_mode: Callable = operator.ge,
+    threshold_dim: str = "threshold",
+) -> XarrayLike:  # type: ignore
+    """
+    Calculates the Brier score for an ensemble forecast for the provided thresholds. By default,
+    the fair Brier score is calculated, which is a modified version of the Brier score that
+    applies a correction related to the number of ensemble members and the number of
+    ensemble members that predict the event.
+
+    Given an event defined by ``event_threshold``, the fair Brier score for ensembles is defined as
+
 
     .. math::
-        \\text{brier score} = \\frac{1}{n} \\sum_{i=1}^n (\\text{forecast}_i - \\text{observed}_i)^2
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2 - \\frac{i(m-i)}{m^2(m-1)}
+
+    Where:
+
+    - :math:`i` is the number of ensemble members that predict the event
+    - :math:`m` is the total number of ensemble members
+    - :math:`y` is the binary observation (0 if the event was not observed or 1 otherwise)
+
+    When the fair correction is not applied (i.e., ``fair_correction=False``),
+    the Brier score is calculated as:
 
-    If you want to speed up performance with Dask and are confident of your input data,
-    or if you want observations to take values between 0 and 1, set ``check_args`` to ``False``.
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2
+
+    If ``fair_correction`` is set to ``True`` and the number of ensemble members is 1, the
+    fair correction will not be applied to avoid division by zero. If you want to
+    set the minimum number ensemble members required to calculate the Brier score,
+    we recommend preprocessing your data to remove data points with fewer than the
+    minimum number of ensemble members that you want.
 
     Args:
-        fcst: Forecast or predicted variables in xarray.
-        obs: Observed variables in xarray.
+        fcst: Real valued ensemble forecasts in xarray. These values are converted to the
+            probability of the event occurring in this function.
```

Comment:
A thought, which probably doesn't matter. Re. the sentence "These values are converted to the probability of the event occurring in this function.":

When I initially read this sentence, I had to momentarily think about the sentence construction to figure out the function must convert values to the probability of the event occurring, rather than the values are converted into the probability of the event occurring within the function (as the latter meaning wouldn't make sense).

Would it be clearer to perhaps say:
"This function converts the values to the probability of the event occurring."
or
"These values are converted to the probability of the event occurring by this function."?

Reply:
The reword seems fine to me, but also somewhat extraneous. Would a user particularly care whether the values are converted to binary inside the implementation? Perhaps it's helpful but from a coding perspective, that's implementation business and I don't know how useful it is to mention.

Reply:
The Brier score is typically defined for when the forecast value is the probability of an event. 

In this case, the probabilities are calculated within the implementation, so this implementation of the Brier score is actually some processing of ensemble data to probabilities and then the Brier score proper.

I requested Nick to make clear what the nature of the forecast inputs are, and that is it different from what some scientific users might expect who are familiar with the Brier

Reply:
So in this case I don't feel that the wording is extraneous

Reply:
No problem, thanks for the explanation.